### PR TITLE
Compiler warnings fix

### PR DESCRIPTION
**Pull request by devmitch**
This PR fixes almost all compiler warnings in the `develop` branch including the Rcpp and Boost warnings from #311. The only remaining warning is
```
 ff16w_strategy.cpp: In member function ‘virtual double plant::FF16w_Strategy::net_mass_production_dt(const plant::FF16_Environment&, double, double, bool)’:
 ff16w_strategy.cpp:30:16: warning: unused variable ‘water_’ [-Wunused-variable]
    30 |   const double water_ = water_access(environment, height, area_leaf_);
```
which I think is fine as the file doesn't seem to be fully implemented yet. It would be good to get a sanity check that the compilation process still works on other machines.

### CODE REVIEW

**Comment by dfalster** [comment]
Hi @devmitch 

Nice work! I love quite compilers. Thank you!

> It would be good to get a sanity check that the compilation process still works on other machines.

Works on my mac!

I've also just approved you as developer on this, so when you send a PR it will trigger automated tests to run via Github Actions. (FYI- These tests are stored in [.github/worklfows](https://github.com/traitecoevo/plant/tree/develop/.github/workflows). The R-CMD-check tests compilation and runs tests on 3 different architectures, so if tests pass here you can pretty confident about functionality on other machines: https://github.com/traitecoevo/plant/blob/develop/.github/workflows/R-CMD-check.yaml#L28

I agree with your take on the one remaining warning -- this is only partially implemented.

**Review by aornugent — APPROVED** [review]
Great work, thank you! Our typical workflow is that PR authors merge once their changes have been approved, preferably using a squash-commit to keep a tidy commit history. Github handles this nicely and aggregates the smaller changes into the squash commit message, which typically gives a nice summary of changes after some light editing.

**Comment by dfalster** [comment]
Hi @devmitch  - I just invited you to join plant team, which should give you write access for this repo. 

Please go ahead and merge this into develop branch, using `squash & merge`. Also include good commit message linking to relevant issues, like you did above.

**Pull request merged**
